@here/harp-geometry/lib/ClipLineString.ts

```diff
@@ -153,3 +154,101 @@ export function clipLineString(
 
     return lines;
 }
+
+/**
+ * The result of wrapping a line string.
+ */
+interface WrappedLineString {
+    left: GeoCoordinates[][];
+    middle: GeoCoordinates[][];
+    right: GeoCoordinates[][];
+}
+
+/**
+ * Helper function to wrap a line string projected in web mercator.
+ *
+ * @param multiLineString The input to wrap
+ * @param edges The clipping edges used to wrap the input.
+ * @param offset The x-offset used to displace the result
+ *
+ * @internal
+ */
+function wrapMultiLineStringHelper(
+    multiLineString: Vector2[][],
+    edges: ClipEdge[],
+    offset: number
+): GeoCoordinates[][] | undefined {
+    for (const clip of edges) {
+        multiLineString = clip.clipLines(multiLineString);
+    }
+
+    const worldP = new Vector3();
+
+    const coordinates: GeoCoordinates[][] = [];
+
+    multiLineString.forEach(lineString => {
+        if (lineString.length === 0) {
+            return;
+        }
+
+        const coords = lineString.map(({ x, y }) => {
+            worldP.set(x, y, 0);
+            const geoPoint = webMercatorProjection.unprojectPoint(worldP);
+            geoPoint.longitude += offset;
+            return geoPoint;
+        });
+
+        coordinates.push(coords);
+    });
+
+    return coordinates.length > 0 ? coordinates : undefined;
+}
+
+/**
+ * Wrap the given line string.
+ *
+ * @remarks
+ * This function splits this input line string in three parts.
+ *
+ * The `left` member of the result contains the part of the line string with longitude less than `-180`.
+ *
+ * The `middle` member contains the part of the line string with longitude in the range `[-180, 180]`.
+ *
+ * The `right` member contains the part of the line string with longitude greater than `180`.
+ *
+ * @param coordinates The coordinates of the line string to wrap.
+ */
+export function wrapLineString(coordinates: GeoCoordinates[]): Partial<WrappedLineString> {
+    const worldP = new Vector3();
+
+    const lineString = coordinates.map(g => {
+        const { x, y } = webMercatorProjection.projectPoint(g, worldP);
+        return new Vector2(x, y);
+    });
+
+    const multiLineString = [lineString];
+
+    return {
+        left: wrapMultiLineStringHelper(multiLineString, WRAP_LEFT_CLIP_EDGES, 360),
+        middle: wrapMultiLineStringHelper(multiLineString, WRAP_MIDDLE_CLIP_EDGES, 0),
+        right: wrapMultiLineStringHelper(multiLineString, WRAP_RIGHT_CLIP_EDGES, -360)
+    };
+}
+
+const ec = EarthConstants.EQUATORIAL_CIRCUMFERENCE;
+const border = 0;
```

Comment:
the default border is 0 for now, I need to understand what to do with borders for these multi line strings, but if it turns out that we need to add borders we can do that in a follow up PR.